app/src/main/java/com/github/se/eduverse/viewmodel/PdfConverterViewModel.kt

```diff
@@ -1,252 +1,145 @@
-      // kept to be safe
-      else -> {
-        context.showToast("Unsupported file type")
-      }
-    }
+  fun setNewFileName(pdfFileName: String) {
+    // Sanitize the file name to prevent malicious file names
+    val sanitizedFileName = pdfFileName.replace(Regex("[^a-zA-Z0-9._-]"), "_")
+    _newFileName.value = sanitizedFileName
   }
 
   /**
-   * Convert the text file corresponding to the given URI to a PDF document
+   * Handle the generation of a PDF file from the given URI using the specified converter option
    *
-   * @param fileUri The URI of the text file to convert
+   * @param uri The URI of the file to convert to PDF
    * @param context The context of the application
+   * @param converterOption The option to use to convert the file to PDF
    */
-  fun convertTextToPdf(fileUri: Uri, context: Context) {
-    viewModelScope.launch(Dispatchers.IO) {
-      try {
-        // Open the text file input stream and create a buffered reader on it, display a toast if
-        // the file fails to open
-        val inputStream = context.contentResolver.openInputStream(fileUri)
-        val bufferedReader =
-            inputStream?.bufferedReader()
-                ?: run {
-                  withContext(Dispatchers.Main) {
-                    Log.e("convertTextToPdf", "Failed to open text file")
-                    context.showToast("Failed to open text file")
-                  }
-                  return@launch
-                }
-
-        // Create a new PDF document
-        val pdfDocument = PdfDocument()
-        val pageInfo = PdfDocument.PageInfo.Builder(595, 842, 1).create() // A4 page size
-        var page = pdfDocument.startPage(pageInfo)
-        var canvas = page.canvas
-        val paint =
-            Paint()
-                .apply { // Set the paint properties for the text to be written on the pdf document
-                  isAntiAlias = true
-                  textSize = 12f
-                  color = Color.BLACK
+  fun generatePdf(uri: Uri?, context: Context, converterOption: PdfConverterOption) {
+    _pdfGenerationState.value = PdfGenerationState.InProgress
+    pdfGenerationJob =
+        viewModelScope.launch {
+          try {
+            val pdfFile: PdfDocument =
+                when (converterOption) {
+                  PdfConverterOption.IMAGE_TO_PDF -> pdfRepository.convertImageToPdf(uri, context)
+                  PdfConverterOption.TEXT_TO_PDF -> pdfRepository.convertTextToPdf(uri, context)
+                  PdfConverterOption.DOCUMENT_TO_PDF -> throw Exception("Not implemented")
+                  PdfConverterOption.SUMMARIZE_FILE -> throw Exception("Not implemented")
+                  PdfConverterOption.EXTRACT_TEXT -> throw Exception("Not implemented")
+                  PdfConverterOption.NONE ->
+                      throw Exception("No converter option selected") // Should never happen
                 }
```

Comment:
Consider adding a loading indicator or a UI state reflecting the "InProgress" state to provide feedback during PDF generation.

Reply:
It's already done